apps/bottle/src/store/query/ServerFetchBoundary.tsx

```diff
@@ -0,0 +1,19 @@
+import { FetchQueryOptions, HydrationBoundary, QueryClient, dehydrate } from '@tanstack/react-query';
+import type { ReactNode } from 'react';
+
+export type FetchOptions = Pick<FetchQueryOptions, 'queryKey' | 'queryFn'>;
+
+type Props = {
+  fetchOptions: FetchOptions[] | FetchOptions;
+  children: ReactNode | ReactNode[];
+};
+
+export async function ServerFetchBoundary({ fetchOptions, children }: Props) {
+  const queryClient = new QueryClient();
```

Comment:
What do you think about managing queryClient with useState? It's intended to ensure it's referenced only once in React's lifecycle. I'm curious about your opinion.

Reply:
Actually,`ServerFetchBoundary` is a ServerComponent and the code runs ONLY on the server. So, I'm not allowed to use `useState` here and it is necessary to make a new QueryClient every time on the server. 
By the way, I think it is there is a better way to make this work. I fixed my code based on the document.
https://tanstack.com/query/latest/docs/framework/react/guides/advanced-ssr#streaming-with-server-components

Reply:
[5324a7b](https://github.com/Nexters/Bottles_Web/pull/38/commits/5324a7b0e997c5a8d5046d6932860656d822ac56)

Reply:
I didn't realize it was a server component..!😅 The changes look good!